src/svg.rs

```diff
@@ -33,13 +35,37 @@ impl SvgVersion {
 declare_surface!(SvgSurface, SurfaceType::Svg);
 
 impl SvgSurface {
-    pub fn new<P: AsRef<Path>>(width: f64, height: f64, path: P) -> Result<SvgSurface, Status> {
-        let path = path.as_ref().to_string_lossy().into_owned();
-        let path = CString::new(path).unwrap();
+    pub fn new<P: AsRef<Path>>(
+        width: f64,
+        height: f64,
+        path: Option<P>,
+    ) -> Result<SvgSurface, Status> {
+        #[cfg(not(windows))]
+        let path = {
+            let path = path.map(|p| {
+                CString::new(p.as_ref().as_os_str().as_bytes())
+                    .expect("Invalid path with NULL bytes")
+            });
+            path.map(|p| p.as_ptr()).unwrap_or(ptr::null())
+        };
+        #[cfg(windows)]
+        let path = {
+            // Based on GLib path to C
+            let path_str = path
```

Comment:
This assumes that `path` is not an `Option` and already a `Path`. Needs the `map()`, `as_ref()` part from above too